pkg/github/comment.cue

```diff
@@ -70,3 +76,119 @@ UpdateComment :: {
         }
     }
 }
+
+Comment :: {
+    subjectId: string
+    marker: *"<!-- bl-marker-do-not-remove -->" | string
+    token: bl.Secret
+    body: string
+
+    listComments: Query & {
+        "token": token
+
+        query:
+            """
+            query($nodeId: ID!) {
+                node(id: $nodeId) {
+                    ...PullRequestParts
+                }
+            }
+            fragment PullRequestParts on PullRequest {
+                id
+                comments(first: 100) {
+                    nodes {
+                        ...CommentParts
+                    }
+                }
+            }
+            \(CommentFragment)
+            """
+        variable: {
+            nodeId: subjectId
+        }
+    }
+
+    // Contains a list of comment ID matching the marker
+    commentId: [n.id for n in listComments.data.node.comments.nodes if strings.Contains(n.body, "\(marker)")]
+
+    updateCommentQuery: json.Marshal({
+        query: """
+            mutation ($input: UpdateIssueCommentInput!) {
+                updateIssueComment(input: $input) {
+                    issueComment {
+                        ...CommentParts
+                    }
+                }
+            }
+            \(CommentFragment)
+            """
+        variables: input: {
+            if len(commentId) > 0 {
+                id: commentId[0]
+            }
+            "body": "\(body)\n\(marker)"
+        }
+    })
+
+    addCommentQuery: json.Marshal({
```

Comment:
nit: this is an unnecessary copy. Instead, you could do:

```
input: "/addCommentQuery": AddComment.query
input: "/updateCommentQuery": UpdateComment.query
```

Reply:
updated based on feedback